dojo/tools/drheader/parser.py

```diff
@@ -11,46 +10,74 @@ def get_label_for_scan_types(self, scan_type):
+        return "Import DrHeader JSON output."
+
+    @staticmethod
+    def _create_finding(test, finding, url=None):
+        rule = finding.get("rule")
+        value = finding.get("value", "")
+        expected = finding.get("expected", [])
+        anomalies = finding.get("anomalies", [])
+        delimiter = finding.get("delimiter", ", ")
+
+        title = f"Header: {rule}"
+        description = [f"{finding['message']}"]
 
-    def return_finding(self, test, finding, url=None):
-        title = "Header : " + finding["rule"]
         if url is not None:
-            message = finding["message"] + "\nURL : " + url
-        else:
-            message = finding["message"]
-        if finding.get("value") is not None:
-            message += "\nObserved values: " + finding["value"]
-        if finding.get("expected") is not None:
-            message += "\nExpected values: "
-            for expect in finding["expected"]:
-                if expect == finding["expected"][-1]:
-                    message += expect
-                else:
-                    message += expect + "; "
-        severity = finding["severity"].title()
-        find = Finding(title=title,
-                    test=test,
-                    description=message,
-                    severity=severity,
-                    static_finding=False)
+            description.append(f"**URL**: {url}")
+
+        if value:
+            description.append(f"**Observed**: {value}")
+
+        if expected:
+            description.append(f"**Expected**: {delimiter.join(expected)}")
+
+        if anomalies:
+            description.append(f"**Anomalies**: {delimiter.join(anomalies)}")
+
+        severity = finding.get("severity").title()
+
+        finding = Finding(
+            title=title,
+            test=test,
+            description="\n".join(description),
+            severity=severity,
+            active=True,
+            verified=True,
+            static_finding=False,
+        )
+
         if url is not None:
-            find.unsaved_endpoints = [Endpoint.from_uri(url)]
-        return find
+            finding.unsaved_endpoints = [Endpoint.from_uri(url)]
 
-    def get_findings(self, filename, test):
-        items = []
+        return finding
+
+    @staticmethod
+    def _parse_json(filename):
         try:
             data = json.load(filename)
-        except ValueError as err:
-            data = {}
-        if data != {} and data[0].get("url") is not None:
-            for item in data:
-                url = item["url"]
-                for finding in item["report"]:
-                    items.append(self.return_finding(test=test, finding=finding, url=url))
-            return items
-        else:
-            for finding in data:
-                items.append(self.return_finding(test=test, finding=finding))
+        except ValueError:
+            data = []  # Workaround for DrHeader invalid json. Issue #8281
+        return data
```

Comment:
Could you get ride of this method and just use `json.load`? silently removing error for malformed reports can have side effects.

Reply:
Sure, I'll remove it.
I left it as is to avoid altering the current functionality for handling the empty report generated by DrHeader. (Issue #8281)

Reply:
Removing the method mentioned above would also require removing the test 'test_parse_file_has_no_findings' in order for tests to pass. Is that what you want? @damiencarol